src/govuk/components/character-count/character-count.test.js

```diff
@@ -2,12 +2,21 @@
  * @jest-environment puppeteer
  */
 
+const jestPuppeteerConfig = require('../../../../jest-puppeteer.config.js')
 const { getExamples } = require('../../../../lib/jest-helpers.js')
 const { goToComponent, renderAndInitialise } = require('../../../../lib/puppeteer-helpers')
 
+// Detect when browser has been launched headless
+const { launch: { headless = true } } = jestPuppeteerConfig
+
 // The longest possible time from a keyboard user ending input and the screen
 // reader counter being updated: handleFocus interval time + last input wait time
-const debouncedWaitTime = 1500
+// but raised to the nearest whole second when browser is not headless as timers
+// in background pages will be aligned, clamped or throttled
+const debouncedWaitTime = headless ? 1500 : 2000
+
+// To avoid typing lag, keyup events are also throttled automatically
```

Comment:
The different wait time is pretty clear, but I'm not understanding this explanation about throttling the keyup events. Why do they need to be throttled when the browser is not headless?

Reply:
Argh, damn. Yeah, I think a better description describing that Chrome messes with `keyup` events under load would help future us 😄 

I'm also wondering if we shouldn't move the component out of using keyboard events for detecting inputs and use the [`input` Event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/input_event). I believe this'll have to wait the work on browser support though as [that event is not supported in IE8](https://caniuse.com/?search=input%20event 😢 )

Reply:
@romaricpascal @36degrees How's this new explanation?

```mjs
// When headless, keydown-to-keyup time appears to affect event throttling,
// affecting `lastInputTimestamp` and screen reader status message updates
const keyupWaitTime = headless ? 0 : 50
```

I'm not 100% on the explanation as perhaps it's related to the whole string of key presses taking less than 500ms, affecting the `this.lastInputTimestamp` **Date.now()** check etc?

Reply:
@romaricpascal Yeah `input` is very interesting. It's probably the fact that the older _Input Events_ (including `keyup`) are technically "cancellable" so for performance maybe they do debounce?